packages/query/src/queriers/app.ts

```diff
@@ -14,19 +12,20 @@ export class AppQuerier implements AppQuerierInterface {
     this.client = createClient(grpcEndpoint, QueryService);
   }
 
-  async appParams() {
-    const req = new AppParametersRequest();
-    const res = await this.client.appParameters(req);
-    return res.appParameters!;
+  async appParams(): Promise<AppParameters> {
+    const { appParameters } = await this.client.appParameters({});
+    if (!appParameters) throw new Error('no app parameters in response');
+    return appParameters;
   }
 
-  async chainParams() {
-    const appParams = await this.appParams();
-    return appParams.chainParams!;
-  }
-
-  async txsByHeight(height: bigint) {
-    const req = new TransactionsByHeightRequest({ blockHeight: height });
-    return await this.client.transactionsByHeight(req);
+  async txsByHeight(blockHeight: bigint): Promise<Transaction[]> {
+    const { blockHeight: responseHeight, transactions } = await this.client.transactionsByHeight({
+      blockHeight,
+    });
+    if (responseHeight !== blockHeight)
```

Comment:
oh dear, is this a real possibility? We get a different block than we requested?

Reply:
idk but it's certainly something that would ruin all of our downstream work